src/k8s-extension/azext_k8s_extension/_params.py

```diff
@@ -99,32 +99,34 @@ def load_arguments(self, _):
         c.argument('force',
                    help='Specify whether to force delete the extension from the cluster.')
 
-    with self.argument_context(f"{consts.EXTENSION_NAME} extension-types list") as c:
+    with self.argument_context(f"{consts.EXTENSION_NAME} extension-types") as c:
         c.argument('cluster_name',
                    options_list=['--cluster-name', '-c'],
                    help='Name of the Kubernetes cluster')
         c.argument('cluster_type',
                    arg_type=get_enum_type(['connectedClusters', 'managedClusters', 'appliances']),
                    options_list=['--cluster-type', '-t'],
                    help='Specify Arc clusters or AKS managed clusters or Arc appliances.')
-
-    with self.argument_context(f"{consts.EXTENSION_NAME} extension-types list-by-location") as c:
-        c.argument('location',
-                   validator=get_default_location_from_resource_group)
-
-    with self.argument_context(f"{consts.EXTENSION_NAME} extension-types show") as c:
-        c.argument('extension_type',
-                   help='Name of the extension type.')
-        c.argument('cluster_name',
-                   options_list=['--cluster-name', '-c'],
-                   help='Name of the Kubernetes cluster')
-        c.argument('cluster_type',
-                   arg_type=get_enum_type(['connectedClusters', 'managedClusters', 'appliances']),
-                   options_list=['--cluster-type', '-t'],
-                   help='Specify Arc clusters or AKS managed clusters or Arc appliances.')
-
-    with self.argument_context(f"{consts.EXTENSION_NAME} extension-types list-versions") as c:
         c.argument('extension_type',
                    help='Name of the extension type.')
         c.argument('location',
-                   validator=get_default_location_from_resource_group)
+                   validator=get_default_location_from_resource_group,
+                   help='Name of the location. Values from: `az account list-locations`')
+        c.argument('version',
+                   help='Version for the extension type.')
+        c.argument('plan_name',
+                   arg_group="Marketplace",
+                   options_list=['--plan-name'],
+                   help='The plan name is referring to the Marketplace Plan ID of the extension.')
+        c.argument('plan_product',
+                   arg_group="Marketplace",
+                   options_list=['--plan-product'],
+                   help='The plan product is referring to the Marketplace Product ID of the extension.')
+        c.argument('plan_publisher',
+                   arg_group="Marketplace",
+                   options_list=['--plan-publisher'],
+                   help='The plan publisher is referring to the Marketplace Publisher ID of the extension')
```

Comment:
Please do not explicitly declare the default option name
```suggestion
        c.argument('plan_name',
                   arg_group="Marketplace",
                   help='The plan name is referring to the Marketplace Plan ID of the extension.')
        c.argument('plan_product',
                   arg_group="Marketplace",
                   help='The plan product is referring to the Marketplace Product ID of the extension.')
        c.argument('plan_publisher',
                   arg_group="Marketplace",
                   help='The plan publisher is referring to the Marketplace Publisher ID of the extension')
```